File: AndroidCustomize/8AutoCompleteTextView/app/src/main/java/com/example/autocompletetextview/MainActivity.java
package com.example.autocompletetextview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        AutoCompleteTextView autoCompleteTextView = findViewById(R.id.autoCompleteTextView);
        String[] coutries = getResources().getStringArray(R.array.countries);
        CountryAdapter countryAdapter = new ArrayAdapter<>(this,R.layout.layout_custom, coutries);
        autoCompleteTextView.setAdapter(arrayAdapter);
    }


    private List<Country> getListCountry(){
        List<Country> list = new ArrayList<>();
        list.add(new Country("Afghanistan"));
        list.add(new Country("Albania"));
        list.add(new Country("Andorra"));
        list.add(new Country("Angola"));
        list.add(new Country("Algeria"));
        list.add(new Country("Argentina"));
        list.add(new Country("Antigua"));
        list.add(new Country("Barbuda"));


        return  list;
    }
}

File: AndroidCustomize/3CustomProgressBar/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ProgressBar


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main);

        ProgressBar progressBar1 = findViewById(R.id.progress_bar_1);
        ProgressBar progressBar2 = findViewById(R.id.progress_bar_2);
        ProgressBar progressBar3 = findViewById(R.id.progress_bar_3);
        ProgressBar progressBar4 = findViewById(R.id.progress_bar_4);
        ProgressBar progressBar5 = findViewById(R.id.progress_bar_5);
        ProgressBar progressBar6 = findViewById(R.id.progress_bar_6);
        ProgressBar progressBar7 = findViewById(R.id.progress_bar_7);
        ProgressBar progressBar8 = findViewById(R.id.progress_bar_8);
        ProgressBar progressBar9 = findViewById(R.id.progress_bar_9);
        ProgressBar progressBar10 = findViewById(R.id.progress_bar_10);
        ProgressBar progressBar11 = findViewById(R.id.progress_bar_11);
        ProgressBar progressBar12 = findViewById(R.id.progress_bar_12);

        progressBar1.setIndeterminateDrawable(new RotatingPlane());
        progressBar2.setIndeterminateDrawable(new DoubleBounce());
        progressBar3.setIndeterminateDrawable(new Wave());
        progressBar4.setIndeterminateDrawable(new WanderingCubes());
        progressBar5.setIndeterminateDrawable(new Pulse	());
        progressBar6.setIndeterminateDrawable(new ChasingDots());
        progressBar7.setIndeterminateDrawable(new ThreeBounce());
        progressBar8.setIndeterminateDrawable(new Circle());
        progressBar9.setIndeterminateDrawable(new CubeGrid());
        progressBar10.setIndeterminateDrawable(new FadingCircle());
        progressBar11.setIndeterminateDrawable(new FoldingCube());
        progressBar12.setIndeterminateDrawable(new RotatingCircle());


    }
}